src/MediatR/Pipeline/Streams/StreamRequestExceptionProcessorBehavior.cs

```diff
@@ -0,0 +1,91 @@
+namespace MediatR.Pipeline.Streams
+{
+    using MediatR.Internal;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+    using System.Runtime.CompilerServices;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Behavior for executing all <see cref="IStreamRequestExceptionHandler{TRequest,TResponse,TException}"/>
+    ///     or <see cref="StreamRequestExceptionHandler{TRequest,TResponse}"/> instances
+    ///     after an exception is thrown by the following pipeline steps
+    /// </summary>
+    /// <typeparam name="TRequest">Request type</typeparam>
+    /// <typeparam name="TResponse">Response type</typeparam>
+    public class StreamRequestExceptionProcessorBehavior<TRequest, TResponse> : IStreamPipelineBehavior<TRequest, TResponse>
+        where TRequest : notnull
+    {
+        private readonly ServiceFactory _serviceFactory;
+
+        public StreamRequestExceptionProcessorBehavior(ServiceFactory serviceFactory) => _serviceFactory = serviceFactory;
+
+        public async IAsyncEnumerable<TResponse> Handle(TRequest request, [EnumeratorCancellation] CancellationToken cancellationToken, StreamHandlerDelegate<TResponse> next)
+        {
+            var asyncEnum = next().WithCancellation(cancellationToken).ConfigureAwait(false).GetAsyncEnumerator();
+
+            bool thrown = false;
+            StreamRequestExceptionHandlerState<TResponse> state = new StreamRequestExceptionHandlerState<TResponse>();
```

Comment:
maybe do the ``new()`` only on demand if an exception is thrown?

Reply:
Good catch. Fixed it. Thanks.